File: nand2tetris-go/cmd/vm_translator.go
package cmd

import (
	"github.com/spf13/cobra"

	"github.com/chehsunliu/nand2tetris/nand2tetris-go/pkg/vm"
)

func newVmTranslatorCmd() *cobra.Command {
	var options vm.Options

	cmd := &cobra.Command{
		Use:   "vm-translator",
		Short: "Translate VM code into Hack assembly",
		RunE: func(cmd *cobra.Command, args []string) error {
			return vm.Translate(&options)
		},
	}

	cmd.Flags().StringVar(&options.InputPath, "input-path", "", "Filepath of VM code")
	if err := cmd.MarkFlagRequired("input-path"); err != nil {
		panic(err)
	}

	return cmd
}

File: nand2tetris-rust/extra/VMTranslator.py
import subprocess
import sys


# The source code is on GitHub: https://github.com/chehsunliu/nand2tetris/nand2tetris-rust.
if __name__ == "__main__":
    subprocess.call(["./nand2tetris-rust", "vm-translator", "--input-path", sys.argv[1]])
